app/scripts/controllers/network/network-controller.ts

```diff
@@ -0,0 +1,1171 @@
+  /**
+   * @see {@link NetworkControllerInfuraIsUnblockedEvent}
+   */
+  InfuraIsUnblocked = 'NetworkController:infuraIsUnblocked',
+}
+
+/**
+ * `networkWillChange` is published when the current network is about to be
+ * switched, but the new provider has not been created and no state changes have
+ * occurred yet.
+ */
+type NetworkControllerNetworkWillChangeEvent = {
+  type: NetworkControllerEventType.NetworkWillChange;
+  payload: [];
+};
+
+/**
+ * `networkDidChange` is published after a provider has been created for a newly
+ * switched network (but before the network has been confirmed to be available).
+ */
+type NetworkControllerNetworkDidChangeEvent = {
+  type: NetworkControllerEventType.NetworkDidChange;
+  payload: [];
+};
+
+/**
+ * `infuraIsBlocked` is published after the network is switched to an Infura
+ * network, but when Infura returns an error blocking the user based on their
+ * location.
+ */
+type NetworkControllerInfuraIsBlockedEvent = {
+  type: NetworkControllerEventType.InfuraIsBlocked;
+  payload: [];
+};
+
+/**
+ * `infuraIsBlocked` is published either after the network is switched to an
+ * Infura network and Infura does not return an error blocking the user based on
+ * their location, or the network is switched to a non-Infura network.
+ */
+type NetworkControllerInfuraIsUnblockedEvent = {
+  type: NetworkControllerEventType.InfuraIsUnblocked;
+  payload: [];
+};
+
+/**
+ * The set of events that the NetworkController messenger can publish.
+ */
+type NetworkControllerEvent =
+  | NetworkControllerNetworkDidChangeEvent
+  | NetworkControllerNetworkWillChangeEvent
+  | NetworkControllerInfuraIsBlockedEvent
+  | NetworkControllerInfuraIsUnblockedEvent;
+
+/**
+ * The messenger that the NetworkController uses to publish events.
+ */
+type NetworkControllerMessenger = RestrictedControllerMessenger<
+  typeof name,
+  never,
+  NetworkControllerEvent,
+  never,
+  NetworkControllerEventType
+>;
+
+/**
+ * Information used to set up the middleware stack for a particular kind of
+ * network. Currently has overlap with `NetworkConfiguration`, although the
+ * two will be merged down the road.
+ */
+type ProviderConfiguration = {
+  /**
+   * Either a type of Infura network, "localhost" for a locally operated
+   * network, or "rpc" for everything else.
+   */
+  type: ProviderType;
+  /**
+   * The chain ID as per EIP-155.
+   */
+  chainId: ChainId;
+  /**
+   * The URL of the RPC endpoint. Only used when `type` is "localhost" or "rpc".
+   */
+  rpcUrl?: string;
+  /**
+   * The shortname of the currency used by the network.
+   */
+  ticker?: string;
+  /**
+   * The user-customizable name of the network.
+   */
+  nickname?: string;
+  /**
+   * User-customizable details for the network.
+   */
+  rpcPrefs?: {
+    blockExplorerUrl?: string;
+  };
+  /**
+   * The ID of the network configuration associated with this provider
+   * configuration.
+   */
+  id?: NetworkConfigurationId;
```

Comment:
@adonesky1 Do you know if this property is actually used? I swear I saw it being set at some point but now I can't find it.

Reply:
It has definitely never been used as "provider configuration". Our middleware stack doesn't use this ID for anything.

Reply:
Okay, I've removed this for now: [de0c195](https://github.com/MetaMask/metamask-extension/pull/18358/commits/de0c195d6a1633f3af189be91b548e5e669f9af2). Happy to add this back later though if we need it.